### PR TITLE
LPS-182597 | Test Improvement | Notifications

### PR DESCRIPTION
**Pull request by JhossephG**
[LPS-182597](https://issues.liferay.com/browse/LPS-182597), [LPS-182505](https://liferay.atlassian.net/browse/LPS-182505), [LPS-194536](https://liferay.atlassian.net/browse/LPS-194536#), [LPS-194535](https://liferay.atlassian.net/browse/LPS-194535#)

### CODE REVIEW

**Comment by JhossephG** [comment]
ci:test:sf

**Comment by JhossephG** [comment]
ci:test:sf

**Comment by timpak** [comment]
Taking a look. This is going to take a little while to review.

**Comment by JhossephG** [comment]
Fine, I'll just add a BDD task that is missing, I just noticed it now

**Comment by timpak** [comment]
Hey @JhossephG there's a merge conflict. Can you resolve this before I continue with my review?

**Comment by JhossephG** [comment]
ci:test:sf

**Comment by JhossephG** [comment]
I'm facing some issues running my SF locally, when I fix it I'll update the PR again.

**Comment by timpak** [comment]
If you can't resolve the SF issues by tomorrow morning, please bring it up in our channel

**Comment by JhossephG** [comment]
ci:test:sf

**Comment by JhossephG** [comment]
It's ready @timpak

**Comment by timpak** [comment]
Got it, don't forget to remove the `Author Action Required` label and assign the PR to me next time!

**Comment by JhossephG** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 1]
```diff
@@ -403,44 +403,100 @@ definition {
...
 				name = "Notification Template",
 				notificationType = "Email",
-				subject = "Subject");
+				subject = "Subject Field");
 
 			PortletEntry.save();
 		}
 
-		task ("When the user edits the template") {
+		task ("When a custom object with a custom field are created") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:511]: and Given a custom object definition

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 2]
```diff
@@ -403,44 +403,100 @@ definition {
...
+				fieldLabelName = "Custom Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject161918");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject161918");
+		}
+
+		task ("And when a action on after add to send a email are created") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:528]: and Given an action to send the email notification after adding an object entry

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 3]
```diff
@@ -403,44 +403,100 @@ definition {
...
+				active = "true",
+				inputMethod = "true",
+				inputValue = "Action shot on add",
+				notificationName = "Notification Template",
+				objectLabel = "Custom Object 161918",
+				thenAction = "Notification",
+				whenAction = "On After Add");
+		}
+
+		task ("And when the user edits the template") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:546]: When the user edits the email notification template

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 4]
```diff
@@ -403,44 +403,100 @@ definition {
...
-				externalReferenceCode = "ERC Edit",
-				fromAddress = "Liferay Edit",
-				fromName = "Test Test Edit",
-				name = "New Notification Edit",
-				subject = "Subject Edit");
+				externalReferenceCode = "ERC Edited",
+				fromAddress = "test2@liferay.com",
+				fromName = "Test Test Edited",
+				name = "Notification Template Edited",
+				subject = "Subject Field Edited");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:580]: Move this to after editing the email template

`Then the email notification template is updated`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 5]
```diff
@@ -403,44 +403,100 @@ definition {
...
 			PortletEntry.save();
 		}
 
-		task ("Then the notification template will have all edited values") {
+		task ("And when an entry is added on the custom object") {
+			ObjectAdmin.addObjectSingleFieldEntryViaAPI(
+				fieldName = "customObjectField",
+				objectName = "CustomObject161918",
+				value = "Trigger");
+		}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:570]: and When an object entry is added

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 6]
```diff
@@ -403,44 +403,100 @@ definition {
...
+				fromName = "Test Test Edited",
+				name = "Notification Template Edited",
+				subject = "Subject Field Edited");
+
+			MockMock.viewMail(
+				asset = "Subject Field Edited",
+				content = "Email Body Content Edited",
+				recipient = "test2@liferay",
+				sender = "test2@liferay.com",
+				subject = "Subject Field Edited");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:587]: Then the action is triggered and an email is sent using the updated notification template

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 7]
```diff
@@ -603,31 +606,32 @@ definition {
 			PortletEntry.save();
 		}
 
-		task ("Go to the created Object and add an Entry") {
-			ObjectAdmin.goToCustomObject(objectName = "CustomObject161920");
-
-			ObjectAdmin.addObjectStringFieldEntryViaUI(entryValue = "Entry");
+		task ("On the created Object add a Entry") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:692]: For now this is fine but we should refactor all of the tasks in the future.

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro [hunk 8]
```diff
@@ -180,6 +180,20 @@ definition {
...
+
+		Click(
+			key_optionName = ${entityName},
+			locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+
+		for (var entityTerms : list ${listOfTerms}) {
+			AssertElementPresent(
+				key_termName = ${entityTerms},
+				locator1 = "ObjectNotifications#ENTITY_TERMS");
+		}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro:236]: A couple of things:
- This is combining two things, a navigation and an assertion. Navigation macros should be their own separate entity so that they can be reused in other areas. For this use case specifically, I'm not sure if there's a lot of value in encapsulating this in a macro unless we have a lot of tests doing this.
- We shouldn't be including an assertion in a macro, that should be on the testcase level

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro [hunk 9]
```diff
@@ -3,57 +3,99 @@ definition {
 	macro addNewNotificationTemplate {
 		LexiconEntry.gotoAdd();
 
-		MenuItem.click(menuItem = ${notificationType});
+		if (${notificationType} == "Email") {
+			MenuItem.click(menuItem = ${notificationType});
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro:7]: Instead of creating a mega macro to cover both, we can just split this into two separate macros. It makes them easier to read and understand

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro [hunk 10]
```diff
@@ -3,57 +3,99 @@ definition {
...
+					locator1 = "TextInput#ANY",
+					value1 = ${cc});
+			}
+
+			if (isSet(bcc)) {
+				Type(
+					key_text = "BCC",
+					locator1 = "TextInput#ANY",
+					value1 = ${bcc});
+			}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro:53]: This should be sorted alphabetically

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 11]
```diff
@@ -201,7 +201,7 @@ definition {
 
 	@description = "LPS-161917 - Verify that it's possible to create an e-mail notification template"
 	@priority = 5
-	test CanCreateEmailTemplate {
+	test CanCreateEmailNotificationTemplate {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:204]: Why is this change part of 6092c70f81e716166f882b70e997533edd337b52?

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro [hunk 12]
```diff
@@ -241,70 +241,100 @@ definition {
...
-				value1 = ${description});
-		}
-
-		if (isSet(entryTo)) {
+		if (${notificationType} == "Email") {
 			AssertTextEquals(
-				key_text = "To",
+				key_text = "Name",
 				locator1 = "TextInput#ANY",
-				value1 = ${entryTo});
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro]: Same thing here, I'm ok with creating two separate macros for each template. There may be some redundancies but it's easier to read.

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro [hunk 13]
```diff
@@ -241,70 +241,100 @@ definition {
...
 
-		if (isSet(subject)) {
+			if (isSet(description)) {
+				AssertTextEquals(
+					key_text = "description",
+					locator1 = "TextArea#ANY",
+					value1 = ${description});
+			}
+
+			if (isSet(entryTo)) {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro:267]: Remember to sort stuff like this alphabetically, unless we need to add them in a specific order

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 14]
```diff
@@ -204,14 +204,14 @@ definition {
 	test CanCreateEmailNotificationTemplate {
 		property portal.acceptance = "true";
 
-		task ("Given the user navigates to Notifications - Templates in the Control Panel") {
+		task ("Given the user navigates to Notifications Templates in the Control Panel") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:207]: This change shouldn't be part of this, it's a wording change so you should use a separate commit with the message `Wordsmith`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 15]
```diff
@@ -204,14 +204,14 @@ definition {
...
 			ApplicationsMenu.gotoPortlet(
 				category = "Notifications",
 				panel = "Control Panel",
 				portlet = "Templates");
 		}
 
-		task ("When the user creates a new Notification Template") {
+		task ("When the user creates a new Email Notification Template") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:214]: Same thing here

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 16]
```diff
@@ -231,7 +231,7 @@ definition {
 			PortletEntry.save();
 		}
 
-		task ("Then the Notification Template is saved with the correct values") {
+		task ("Then the Email Notification Template is saved with the correct values") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:234]: Same here

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 17]
```diff
@@ -250,6 +250,52 @@ definition {
...
+		property portal.acceptance = "true";
+
+		task ("Given the user navigates to Notifications Templates in the Control Panel") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Notifications",
+				panel = "Control Panel",
+				portlet = "Templates");
+		}
+
+		task ("When the user creates a new User Notification Template") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:265]: `When the user adds a new user notification template`

No need to capitalize since it's not a proper noun

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 18]
```diff
@@ -250,6 +250,52 @@ definition {
...
+				userName = "Test");
+
+			ObjectNotifications.selectEntityAndCheckTerms(
+				entityName = "User",
+				listOfTerms = "Middle Name,Screen Name,Create Date,First Name,Last Login Date,UUID");
+
+			PortletEntry.save();
+		}
+
+		task ("Then the User Notification Template is saved with the correct values") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:281]: Then the user notification template is created

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 19]
```diff
@@ -460,6 +460,40 @@ definition {
 		}
 	}
 
+	@description = "LPS-182505 - Verify that it's possible to delete an user notification template"
+	@priority = 3
+	test CanDeleteUserTemplate {
+		task ("Given a User Notification Template") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:460]: Use lower case like in the previous examples

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 20]
```diff
@@ -460,6 +460,40 @@ definition {
...
+
+				ObjectNotifications.addNewNotificationTemplate(
+					description = "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
+					name = "User Notification Template",
+					notificationType = "User Notification",
+					recipientType = "User",
+					subject = "Subject content",
+					userName = "Test");
+
+				PortletEntry.save();
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: We should do this via API instead, also your indentation is off for a few of the macros

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 21]
```diff
@@ -460,6 +460,40 @@ definition {
...
+			}
+
+			task ("When delete the User Notification Template") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: When the user deletes the user notification template

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 22]
```diff
@@ -460,6 +460,40 @@ definition {
...
+				ObjectNotifications.deleteNotificationTemplate(templateName = "Notification Template");
+			}
+
+			task ("Then the User Notification Template is not present") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: Then the user notification template is no longer present in the table

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 23]
```diff
@@ -215,7 +215,7 @@ definition {
 			ObjectNotifications.addNewNotificationTemplate(
 				bcc = "test3@liferay.com",
 				cc = "test2@liferay.com",
-				description = "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
+				description = "Lorem Ipsum is simply dummy text of the printing and typesetting industry.Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: Why is this change part of 913e3f7f1777cfbae83bf83165d45b875aaa189e? It's not a logical part of the changes you're making

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 24]
```diff
@@ -405,14 +402,12 @@ definition {
...
+				fieldBusinessType = "Text",
 				fieldLabelName = "Custom Field",
 				fieldName = "customObjectField",
-				fieldType = "Long",
-				fileSource = "userComputer",
+				fieldType = "String",
 				isRequired = "false",
-				objectName = "CustomObject161922",
-				showFilesInDocumentsAndMedia = "true");
+				objectName = "CustomObject161922");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: Why are you changing the existing field type here?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 25]
```diff
@@ -598,24 +592,23 @@ definition {
 	@description = "LPS-161923 - Verify that it's possible to search for notifications"
 	@priority = 3
 	test CanSearchForNotifications {
-		task ("Go to Templates and add Notification Templates") {
+		task ("Given a Notification Templates") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:594]: Changing the wording shouldn't be part of this commit because it's not part of switching to the API macro

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 26]
```diff
@@ -598,24 +592,23 @@ definition {
...
-
-				PortletEntry.save();
+					subject = "Subject",
+					templateName = "Notification ${templateName}",
+					toAddress = "test@liferay.com");
 			}
 		}
 
-		task ("Search for Liferay") {
+		task ("When search for Liferay") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:610]: Same here

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 27]
```diff
@@ -627,7 +620,7 @@ definition {
 				value1 = "\RETURN");
 		}
 
-		task ("Assert only Notification Liferay is present") {
+		task ("Then assert only Notification Liferay is present") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:622]: Here too

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 28]
```diff
@@ -761,7 +748,7 @@ definition {
 			ObjectAdmin.addObjectSingleFieldEntryViaAPI(
 				fieldName = "customObjectField",
 				objectName = "CustomObject178781",
-				value = "Entry Test");
+				value = "Trigger");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:750]: Here too

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 29]
```diff
@@ -407,7 +407,7 @@ definition {
 				fieldName = "customObjectField",
 				fieldType = "String",
 				isRequired = "false",
-				objectName = "CustomObject161922");	
+				objectName = "CustomObject161922");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:410]: The changes in e4ddc8ba1c2eda83faca5af287ce5e99790994f9 should be made in the commits that caused them and not in a separate commit like this. Essentially, you're making a mistake in a commit and using another commit to fix it. It's better to go back and fix the original commit so that you never made the mistake in the first place.

**Review by timpak — CHANGES_REQUESTED** [review]
@JhossephG please see my comments below. Also, please change your commit messages to follow existing patterns in our codebase.

The first word shouldn't be a gerund, like `Refactoring` it should just be `Refactor`. You don't need to put `Part: X` as well, since the ticket ID will group your changes logically. 

Don't forget to include the rest of the tickets in your original comment, it only has LPS-182597 right now.

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro (hunk 8 above)
Comment (JhossephG) [reply modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro:236]: As I remember I create this macro to turn easy ensure that the criterias from [this](https://liferay.atlassian.net/browse/LPS-171625) story was been done. I have create it instead put directly on the testcase level to turn reusable in future tests and make easier for whoever was going to do it. So, should I:

1. Delete the macro and put the structure in the testcase level
2. Remove the assertion from the macro and put in testcase level
3. Leave as is
cc:@timpak

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase (hunk 11 above)
Comment (JhossephG) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:204]: I put on [66c8157833893e45385de72ed5386d0f90197590](https://github.com/liferay-bpm-qa/liferay-portal/pull/828/commits/66c8157833893e45385de72ed5386d0f90197590) since it's in the same context.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase (hunk 20 above)
Comment (JhossephG) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: Our API macro for templates just work to email templates. Should I add it on this PR or do in the future?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase (hunk 24 above)
Comment (JhossephG) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: Because the macro **addObjectSingleFieldEntryViaJSONAPI** that I'm using later on don't support attachment fields.

**Comment by JhossephG** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro (hunk 8 above)
Comment (timpak) [reply modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro:247]: If that macro is only being used in one test, #1. If the test is used in multiple tests, then #2.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 30]
```diff
@@ -804,7 +791,7 @@ definition {
 		task ("When the object entry is deleted") {
 			ObjectAdmin.goToCustomObject(objectName = "CustomObject178781");
 
-			ObjectPortlet.deleteEntry(entry = "Trigger");
+			ObjectPortlet.deleteEntry(entry = "Entry test");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: This change doesn't belong here, it should be taken care of in the commit that originally made the change. 
1. You shouldn't be fixing an error you made earlier in another commit, just fix the original commit
2. You shouldn't be adding changes that aren't part of the logical changes in the commit. Since you're fixing the API usage here, you shouldn't be renaming the entry unless it's part of switching to the API.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 31]
```diff
@@ -1060,7 +1035,7 @@ definition {
 
 			ObjectPortlet.updateEntry(
 				entry = "Entry Test",
-				updateEntry = "Trigger");
+				updateEntry = "Entry test");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: Same thing here, see previous comment.

Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: Same thing here, see previous comment.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 32]
```diff
@@ -1632,7 +1600,7 @@ definition {
 
 			ObjectPortlet.viewEntryDetails(entry = "Entry Test");
 
-			ObjectAdmin.editEntryName(newEntryValue = "Trigger");
+			ObjectAdmin.editEntryName(newEntryValue = "Entry Test Edited");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: Same thing here, see previous comment.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase (hunk 29 above)
Comment (timpak) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: @JhossephG let me try explaining this in another way. This commit, 25de24effa51100052d0f58150b2f5b2569d6127, shouldn't exist. All of the changes in this commit are to fix mistakes made in previous commits. You should go back and fix the original commit instead of using a new commit.

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro [hunk 33]
```diff
@@ -325,7 +325,7 @@ definition {
 		Click(
 			key_name = ${name},
 			locator1 = "ObjectAdmin#TABLE_LIST_TITLE");
-		
+
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/macros/ObjectNotifications.macro:328]: This error should be fixed in the commit that caused this.

**Review by timpak — CHANGES_REQUESTED** [review]
@JhossephG please see my comments below.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase (hunk 29 above)
Comment (timpak) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase]: Don't just fix this error seen in the comment, please go and fix all of the errors in this commit in the original commits.

**Comment by JhossephG** [comment]
ci:test:sf

**Comment by JhossephG** [comment]
ci:test:object

**Comment by JhossephG** [comment]
ci:test:object:nocompile

**Comment by JhossephG** [comment]
It's ready again @timpak

**Review by timpak — COMMENTED** [review]
@JhossephG can you check the test results? If the modified tests all pass, please forward this.

**Comment by timpak** [comment]
You can also assign the tickets to yourself and close them out once you forward this.

**Comment by JhossephG** [comment]
ci:test:object:nocompile

**Comment by JhossephG** [comment]
Only this testcase CanSendEmailViaActionDelete is failing cause a mistake that I did choosing the entry value content, I'll update this.

**Comment by JhossephG** [comment]
ci:test:object:nocompile

**Comment by JhossephG** [comment]
Ok, seeing the results all the tests that I changed something are passing

**Comment by JhossephG** [comment]
ci:forward
